alacritty/src/display/mod.rs

```diff
@@ -188,9 +188,17 @@ pub struct Display {
 
 impl Display {
     pub fn new<E>(config: &Config, event_loop: &EventLoop<E>) -> Result<Display, Error> {
-        // Guess DPR based on first monitor.
-        let estimated_dpr =
-            event_loop.available_monitors().next().map(|m| m.scale_factor()).unwrap_or(1.);
+        #[cfg(any(not(feature = "x11"), target_os = "macos", windows))]
+        let is_x11 = false;
+        #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+        let is_x11 = event_loop.is_x11();
```

Comment:
I didn't actually notice at first that this check was needed later anyways. That kinda makes this solution relatively clean.

Reply:
What exactly does this do?

Reply:
@nogamike76 for linux builds (aka. `not(any(target_os = "macos", windows))` builds) with the `x11` feature enabled, it returns `true` when the terminal window is running on X11 as opposed to Wayland.